Add contact information section and its edit subpage to the settings page

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Partial of #863 and based on #918

- Add URLs to utils for settings and edit contact info pages
- Implement the preview section and edit subpage of contact information, and add them to the settings page

💡 Please note that this PR doesn't include the API integration.

### Screenshots:

#### 🎥 . When the phone number got from Google MC is empty

https://user-images.githubusercontent.com/17420811/127619456-9d5d8c83-d353-43e5-be62-0a8bbd1ed803.mp4

#### 🎥 . When the phone number got from Google MC has already exist

https://user-images.githubusercontent.com/17420811/127619741-ae09b9fa-dd57-4378-8950-840fdf131f5d.mp4

### Detailed test instructions:

For easier testing, there is a mock phone number that is 50% chance for an empty phone number and 50% for an existed phone number when every time reload the page.

https://github.com/woocommerce/google-listings-and-ads/blob/4f5eb4e9cb8d13f943c164bb9e1d0b1d22ab84f5/js/src/data/selectors.js#L71-L78

1. Go to the settings page.
2. The "Contact information" should show up with
    - a warning section that prompts users to add information if the phone data is empty.
    - a folded section that displays phone number and store address if the phone data is not empty.
3. Click on "Add information" or "Edit" button, and it should bring you to contact info edit page.
4. In the contact info edit page,
    - the implementations are the same as the previous PRs. Please refer to #917 and #918 if want to test them further.
    - click on "<" icon at the top-left side, and it should bring you back to settings page.

### Changelog entry

eason9487 committed:
Add URLs to utils for settings and edit contact info pages

eason9487 committed:
Implement ContactInformationPreview for the main page of settings

eason9487 committed:
Implement the EditContactInformation page

eason9487 committed:
Add ContactInformationPreview and EditContactInformation to settings page

j111q commented:
One note - we tweaked the wording on the message here:

![image](https://user-images.githubusercontent.com/65582817/127623417-661db0ba-e974-410a-8397-a4d85272fa36.png)

So it should say `Google requires the phone number and store address for all stores using Google Merchant Center. `

We took out the phrase about "owner of the business" because that could be confusing - it's not necessarily the owner who is setting up the integration or managing it.

eason9487 committed:
Update wording

Address https://github.com/woocommerce/google-listings-and-ads/pull/921#issuecomment-889723783

eason9487 commented:
@j111q, thanks for notifying me. Updated by https://github.com/woocommerce/google-listings-and-ads/pull/921/commits/e9de8d5ec3493e4eb6fae425ea9f53c5752ff81b.

![image](https://user-images.githubusercontent.com/17420811/127624147-cd184c49-946f-4d96-9612-ed2fafcc834f.png)

j111q commented:
One more thing - link to documentation / all the "Read more" links should go here: https://docs.woocommerce.com/document/google-listings-and-ads/#contact-information

Thank you @eason9487 !

eason9487 committed:
Update the URL of "Learn more" links

eason9487 commented on js/src/components/contact-information/index.js:
```diff
@@ -22,6 +29,50 @@ const description = __(
+		if ( phone.data.isValid ) {
+			sectionContent = (
+				<VerticalGapLayout size="overlap">
+					<PhoneNumberCard
+						isPreview
+						initEditing={ false }
+						onEditClick={ handleEditClick }
+					/>
```
Dev note: follow the change of https://github.com/woocommerce/google-listings-and-ads/pull/917#discussion_r681154403 to pass `phone` as `phoneNumber` prop before merging this PR.

eason9487 replied:
Added by https://github.com/woocommerce/google-listings-and-ads/pull/921/commits/34e6f62afcd6e1554e3d6df84da5746ba11b0f49.

ecgan commented on js/src/components/contact-information/index.js:
```diff
@@ -22,6 +29,50 @@ const description = __(
+		getHistory().push( getEditContactInformationUrl() );
+	};
+
+	let sectionContent;
```
💅  Perhaps we can set a default `sectionContent` (the one with `AppSpinner`), then we don't need the `else` block in [Line 61-67](https://github.com/woocommerce/google-listings-and-ads/pull/921/files#diff-51ff763be0b20871d193755311c5c437caafd6c4d46bb1413de599e1124a7714R61-R67).

eason9487 replied:
Changed by https://github.com/woocommerce/google-listings-and-ads/pull/921/commits/3dbc1fa23bfd8dbf87823a553668f77f314dcc48.

ecgan commented on js/src/components/contact-information/index.js:
```diff
@@ -22,6 +29,50 @@ const description = __(
+		}
+	} else {
+		sectionContent = (
+			<Section.Card>
+				<AppSpinner />
+			</Section.Card>
```
❔ Would the `SpinnerCard` component be good here?

```suggestion
			<SpinnerCard />
```

eason9487 replied:
The card height will be higher and looks better. Thanks!

![image](https://user-images.githubusercontent.com/17420811/128116657-5bb499de-af8e-4146-9b4e-7c8f7d851e60.png)

eason9487 replied:
Just recalled that the `<Section.Card> <AppSpinner /> </Section.Card>` was taken for aligning the height of all loading cards on the settings page.

![Snapshot 2021-08-04 at 11 22 48](https://user-images.githubusercontent.com/17420811/128120921-8a58d93f-de31-4710-87b4-a3c9296e304d.jpg)

But `<SpinnerCard>` looks better, to make all loading UIs consistency in the Setting page, I adjusted the spinner in the DisconnectAccounts as well - https://github.com/woocommerce/google-listings-and-ads/pull/921/commits/ffba4487bb0a053dc3adb43f4b864c18478bea29

ecgan commented on js/src/components/contact-information/index.js:
```diff
@@ -60,7 +110,10 @@ export default function ContactInformation( { view, onPhoneNumberChange } ) {
 			) : (
-				<AppSpinner />
+				<VerticalGapLayout size="large">
+					<SpinnerCard />
+					<SpinnerCard />
```
Just a note: at first it looks weird to have two `SpinnerCard`s like this, but after digging into the source code, I came to understand they correspond to the `PhoneNumberCard` and `StoreAddressCard` in [Line 106-110](https://github.com/woocommerce/google-listings-and-ads/pull/921/files#diff-51ff763be0b20871d193755311c5c437caafd6c4d46bb1413de599e1124a7714R106-R110).

ecgan commented on js/src/settings/edit-contact-information/index.js:
```diff
@@ -0,0 +1,56 @@
+			/>
+			<div className="gla-settings">
+				<ContactInformation
+					view="settings"
```
❔ The value `"settings"` is not really being used in this PR. I'm guessing it is used in the other PRs? 🤔

eason9487 replied:
Even though the value `"settings"` is not be used, I thought it still good to indicate the `view` prop because I think none of `"settings"` and `"setup-mc"` should be the default value. And I was planning to add JSDos for **ContactInformation** like this:

```js
/**
 * @param {Object} props React props.
 * @param {'setup-mc'|'settings'} props.view
 */
```

ecgan commented on js/src/components/contact-information/no-contact-information-card.js:
```diff
@@ -0,0 +1,59 @@
+							href={ learnMoreUrl }
+						>
+							{ __( 'Learn more', 'google-listings-and-ads' ) }
+						</AppButton>
```
❓ I think we should do track event for this "Learn more" button here. Either we use `AppButton` with `eventName` and `eventProps`, or `AppDocumentationLink` (which I think is better in semantic, but I guess you would need to mingle with CSS layout a little).

eason9487 replied:
Thanks for reminding. Added by https://github.com/woocommerce/google-listings-and-ads/pull/921/commits/bf236cb80fe3a54bc413813424bd17b3cbeaf160.

![image](https://user-images.githubusercontent.com/17420811/128120657-1e4641f1-6b68-4a6b-a2ee-283e28249afc.png)

ecgan reviewed (approved):
Tested as per test instructions, seems to be working good. Left one ❓ comment on external link and track event, and a few ❔ 💅  comments. 

Think you would be able to handle the ❓ comment, approving this now as there isn't any real blocking issues.

eason9487 committed:
Use SpinnerCard instead

Address https://github.com/woocommerce/google-listings-and-ads/pull/921#discussion_r681954844

Co-authored-by: Gan Eng Chin <ecgan@users.noreply.github.com>

eason9487 committed:
Use SpinnerCard instead for the DisconnectAccounts to make all loading UIs consistency in the Setting page

Address https://github.com/woocommerce/google-listings-and-ads/pull/921#discussion_r681954844

eason9487 committed:
Simplify the else condition for ContactInformationPreview

Address https://github.com/woocommerce/google-listings-and-ads/pull/921#discussion_r681952956

eason9487 committed:
Add event tracking for the "Learn more" link in the NoContactInformationCard

Address https://github.com/woocommerce/google-listings-and-ads/pull/921#discussion_r681979584

eason9487 committed:
Pass phone number data as phoneNumber prop to PhoneNumberCard

Address https://github.com/woocommerce/google-listings-and-ads/pull/921#discussion_r681414104
